test/node/object_id_tests.js

```diff
@@ -171,18 +172,22 @@ describe('ObjectId', function () {
     it(`should correctly create ObjectId from ${input} and result in ${output}`, function () {
       const objId = new ObjectId(input);
       expect(objId).to.have.property('id');
-      expect(objId.id).to.be.instanceOf(Buffer);
-      expect(objId.id.readUInt32BE(0)).to.equal(output);
+      expect(isBufferOrUint8Array(objId.id)).to.be.true;
+      const num = new DataView(objId.id.buffer, objId.id.byteOffset, objId.id.byteLength).getInt32(
+        0,
+        false
+      );
+      expect(num).to.equal(output);
     });
   }
 
   it('should correctly create ObjectId undefined or null', function () {
     const objNull = new ObjectId(null);
     const objNoArg = new ObjectId();
     const objUndef = new ObjectId(undefined);
-    expect(objNull.id).to.be.instanceOf(Buffer);
-    expect(objNoArg.id).to.be.instanceOf(Buffer);
-    expect(objUndef.id).to.be.instanceOf(Buffer);
+    expect(isBufferOrUint8Array(objNull.id)).to.be.true;
+    expect(isBufferOrUint8Array(objNoArg.id)).to.be.true;
+    expect(isBufferOrUint8Array(objUndef.id)).to.be.true;
```

Comment:
can we add an expect fail message, since we no longer have an obvious `instanceOf` assertion?

Reply:
Yes, added to all the cases where we use this helper